core/src/main/java/org/elasticsearch/action/index/TransportIndexAction.java

```diff
@@ -208,7 +209,12 @@ private void processAfter(IndexRequest request, IndexShard indexShard, Translog.
         }
 
         if (indexShard.getTranslogDurability() == Translog.Durabilty.REQUEST && location != null) {
-            indexShard.sync(location);
+            try {
+                indexShard.sync(location);
```

Comment:
I think this is the wrong fix since we call this in many places ie. during delete and bulk. I think we should just rename this method into `trySync` and don't sync if the engine is closed. so we handle it correctly everywhere?